src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs

```diff
@@ -64,6 +64,9 @@ protected ExecutionManager(ITestPlatformEventSource testPlatformEventSource, IRe
     /// <param name="pathToAdditionalExtensions"> The path to additional extensions. </param>
     public void Initialize(IEnumerable<string> pathToAdditionalExtensions, ITestMessageEventHandler testMessageEventsHandler)
     {
+        // Clear the request data metrics left over from a potential previous run.
+        _requestData.MetricsCollection.Metrics.Clear();
```

Comment:
On discovery side you are null checking `MetricsCollection` and `Metrics`, shall you do it here too?

Reply:
Fixed, thanks !